internal/executor/utilisation/pod_utilisation.go

```diff
@@ -24,26 +24,39 @@ type PodUtilisationService interface {
 	GetPodUtilisation(pod *v1.Pod) *domain.UtilisationData
 }
 
-type KubeletPodUtilisationService struct {
+type PodUtilisationServiceImpl struct {
 	clusterContext     cluster_context.ClusterContext
+	fetchers           []podUtilisationFetcher
 	nodeInfoService    node.NodeInfoService
 	podUtilisationData map[string]*domain.UtilisationData
 	dataAccessMutex    sync.Mutex
 }
 
-func NewMetricsServerQueueUtilisationService(
+type podUtilisationFetcher interface {
+	fetch(nodes []*v1.Node, podNameToUtilisationData map[string]*domain.UtilisationData, clusterContext clusterContext.ClusterContext)
+}
+
+func NewPodUtilisationService(
```

Comment:
Consistent naming - rename NewMetricsServerQueueUtilisationService to NewPodUtilisationService as that's what it's called elsewhere